velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -339,13 +397,21 @@ HiveDataSink::HiveDataSink(
     sortColumnIndices_.reserve(sortedProperty.size());
     sortCompareFlags_.reserve(sortedProperty.size());
     for (int i = 0; i < sortedProperty.size(); ++i) {
-      sortColumnIndices_.push_back(
-          inputType_->getChildIdx(sortedProperty.at(i)->sortColumn()));
-      sortCompareFlags_.push_back(
-          {sortedProperty.at(i)->sortOrder().isNullsFirst(),
-           sortedProperty.at(i)->sortOrder().isAscending(),
-           false,
-           CompareFlags::NullHandlingMode::kNullAsValue});
+      auto columnIndex =
+          inputType_->getChildIdx(sortedProperty.at(i)->sortColumn());
+      if (std::find(
+              partitionChannels_.cbegin(),
+              partitionChannels_.cend(),
+              columnIndex) == partitionChannels_.cend()) {
+        sortColumnIndices_.push_back(
+            getDataType(dataChannels_, inputType_)
+                ->getChildIdx(sortedProperty.at(i)->sortColumn()));
```

Comment:
```suggestion
auto columnIndex = getDataType(dataChannels_, inputType_)->getChildIdxIfExists(sortedProperty.at(i)->sortColumn()));
if (columnIndex.has_value()) {
    sortColumnIndices_.push_back(columnIndex.value());
}
```
Would the above work and be simpler?